packages/schemas/src/gen/schema.ts

```diff
@@ -33,25 +33,30 @@ export const generateSchema = ({ name, fields }: TableWithType) => {
     '};',
     '',
     `const createGuard: CreateGuard<${databaseEntryType}> = z.object({`,
-    // eslint-disable-next-line complexity
-    ...fields.map(({ name, type, isArray, isEnum, nullable, hasDefaultValue, tsType }) => {
-      if (tsType) {
-        return `  ${camelcase(name)}: ${camelcase(tsType)}Guard${conditionalString(
-          nullable && '.nullable()'
-        )}${conditionalString((nullable || hasDefaultValue) && '.optional()')},`;
-      }
 
-      return `  ${camelcase(name)}: z.${
-        isEnum ? `nativeEnum(${type})` : `${type}()`
-      }${conditionalString(isArray && '.array()')}${conditionalString(
-        nullable && '.nullable()'
-      )}${conditionalString((nullable || hasDefaultValue) && '.optional()')},`;
-    }),
+    ...fields.map(
+      // eslint-disable-next-line complexity
+      ({ name, type, isArray, isEnum, nullable, hasDefaultValue, tsType, stringMaxLength }) => {
+        if (tsType) {
+          return `  ${camelcase(name)}: ${camelcase(tsType)}Guard${conditionalString(
+            nullable && '.nullable()'
+          )}${conditionalString((nullable || hasDefaultValue) && '.optional()')},`;
+        }
+
+        return `  ${camelcase(name)}: z.${
+          isEnum ? `nativeEnum(${type})` : `${type}()`
+        }${conditionalString(stringMaxLength && `.max(${stringMaxLength})`)}${conditionalString(
```

Comment:
As the variable is `stringMaxLength`, will you check whether the previous type is `string` before appending `.max(stringMaxLength)`?

Reply:
Have checked their type before generating schema (`generateSchema`). Only `bpchar`, `char`, `varchar` types will append `.max(stringMaxLength)`.

Reply:
@darcyYe's comment is valid - i don't think combine TWO meanings into ONE property name is a good idea

Reply:
let me tell you how: split to TWO properties - `isString` and `maxLength` (or `max`, whatever)

Reply:
Refactored